agora/Agora/Stake/Scripts.hs

```diff
@@ -301,18 +302,40 @@ stakeValidator stake =
           tcassert "Proposal ST spent" $
             spentProposalST #== 1
 
+          -- Update the stake datum, but only the 'lockedBy' field.
+
+          let -- We actually don't know whether the given lock is valid or not.
+              -- This is checked in the proposal validator.
+              newLock = pfield @"lock" # l
+              -- Prepend the new lock to the existing locks.
+              expectedLocks = pcons # newLock # stakeDatum.lockedBy
+
+          expectedDatum <-
+            tclet $
+              pdata $
+                mkRecordConstr
+                  PStakeDatum
+                  ( #stakedAmount .= stakeDatum.stakedAmount
+                      .& #owner .= stakeDatum.owner
+                      .& #lockedBy .= pdata expectedLocks
+                  )
+
           tcassert "A UTXO must exist with the correct output" $
+            -- FIXME: no need to pass the whole txInfo to 'anyOutput'.
             anyOutput @PStakeDatum # txInfo
```

Comment:
Yep, we should deprecate `anyOutput`, really.

Reply:
I don't know what did you mean by "deprecating" it. If we want to completely remove it, it may be best to open another 
 new ticket. Shouldn't be very complicated though,  it's only used 6 times in our code base:
<img width="347" alt="image" src="https://user-images.githubusercontent.com/40087668/169490218-e73c6aed-d25a-4980-9738-d6bdd5f08d9c.png">

Reply:
Yep. That's what I mean. I'll create a ticket.